src/Microsoft.Sbom.Targets/GenerateSbomTask.cs

```diff
@@ -84,25 +97,36 @@ public override bool Execute()
                 NamespaceUriBase = this.NamespaceBaseUri,
                 NamespaceUriUniquePart = this.NamespaceUriUniquePart,
                 DeleteManifestDirectoryIfPresent = this.DeleteManifestDirIfPresent,
-                Verbosity = ValidateAndAssignVerbosity(),
+                Verbosity = logVerbosity,
             };
 #pragma warning disable VSTHRD002 // Avoid problematic synchronous waits
-            var result = System.Threading.Tasks.Task.Run(() => this.Generator.GenerateSbomAsync(
+            var result = System.Threading.Tasks.Task.Run(() => generator.GenerateSbomAsync(
                 rootPath: this.BuildDropPath,
                 manifestDirPath: this.ManifestDirPath,
                 metadata: sbomMetadata,
                 componentPath: this.BuildComponentPath,
                 runtimeConfiguration: runtimeConfiguration,
                 specifications: ValidateAndAssignSpecifications(),
-                externalDocumentReferenceListFile: this.ExternalDocumentListFile)).GetAwaiter().GetResult();
+                externalDocumentReferenceListFile: this.ExternalDocumentListFile),
+                this.cancellationTokenSource.Token).GetAwaiter().GetResult();
 #pragma warning restore VSTHRD002 // Avoid problematic synchronous waits
 
+            if (!result.IsSuccessful)
+            {
+                Log.LogError("SBOM generation failed. Check the build log for details.");
```

Comment:
This is a user-facing string that tell the user that something went wrong - since I'm not sure what a reasonable summary of the kinds of errors might be, I instead chose to emit details to the MSBuild log at a more verbose log level. Users that need details can check that out.